### PR TITLE
Make meatier more configurable for deployment

### PR DESCRIPTION
**Pull request by patrickleet**
changes to make meatier more configurable through environment variables for deployment - added support for ```DATABASE_HOST```, ```DATABASE_PORT```, ```DATABASE_AUTH_KEY```, and ```DATABASE_SSL``` - ```DATABASE_SSL``` is currently a flag (true/false) which tells the app to load a cert file needed for production rethinkdb deployments that require it. 

Also made ```GRAPHQL_HOST``` and ```GRAPHQL_PROTOCOL``` for configuring graphql. 

Along with these are a few new utils, ```readCert``` in ```src/server/database/readCert.js``` which synchronously reads a cert file (currently from hardcoded location). 

```src/universal/utils/graphQLConfig.js``` - exports two functions that work on client or server: ```getGraphQLHost``` and ```getGraphQLProtocol```. 

```src/server/database/rethink.config.js``` has been replaced with ```src/server/database/getRethinkConfig.js``` which exports a function ```getRethinkConfig``` that deals with reading and including the cert in the config for rethinkdb if it is required. 

Lastly, switched ```npm run start``` to run production, as most places you deploy to expect that, and created ```npm run dev```, which runs in dev mode.

### CODE REVIEW

**Comment by mattkrick** [comment]
this looks *really* good, but since i've been a lazy jerk & haven't added tests for these things I'll need to mark out some time to go through this & make sure it works with SSR, etc. Thanks!

**Comment by patrickleet** [comment]
Cool cool, I also tested it out on the azk branch referenced in #52 (https://github.com/Araphel/meatier)

here is the Azkfile.js if you want to try it out. 

```
systems({
  meatier: {
	  docker_extra: {
      User: 'root',
     },
    depends: ["rethinkdb"],
    // More images:  http://images.azk.io
    image: {"docker": "azukiapp/node"},
    // Steps to execute before running instances
    provision: [
      "npm install",
    ],
    workdir: "/azk/#{manifest.dir}",
    shell: "/bin/bash",
    command: ["npm", "run", "quickstart"],
    wait: {"retry": 2, "timeout": 20000},
    mounts: {
      '/azk/#{manifest.dir}': sync("."),
      '/azk/#{manifest.dir}/node_modules': persistent("./node_modules"),
    },
    scalable: {"default": 1},
    http: {
      domains: [
        "#{env.HOST_DOMAIN}",
        "#{env.HOST_IP}",
        "#{system.name}.#{azk.default_domain}"
      ]
    },
    ports: {
      // exports global variables
      http: "3000/tcp",
    },
    envs: {
      // Make sure that the PORT value is the same as the one
      // in ports/http below, and that it's also the same
      // if you're setting it in a .env file
      NODE_ENV: "development",
      PORT: "3000",
      GRAPHQL_HOST: "#{system.name}.#{azk.default_domain}"
    },
  },

  rethinkdb: {
    image: { docker: "rethinkdb" },
    shell: '/bin/bash',
    scalable: false,
    command: "rethinkdb --bind all --direct-io --cache-size auto --server-name rethinkdb --directory ./rethinkdb --canonical-address rethinkdb.dev.azk.io",
    wait: {"retry": 2, "timeout": 5000},
    mounts: {
      '/rethinkdb': persistent('rethinkdb-#{manifest.dir}'),
    },
    ports: {
      http: "8080",
      data: "28015",
      cluster: "29015",
    },
    http: {
      domains: [ '#{system.name}.#{azk.default_domain}' ],
    },
    export_envs: {
      "DATABASE_HOST": '#{net.host}',
      "DATABASE_PORT": '#{net.port.data}',
      "DATABASE_URL": 'rethinkdb://#{net.host}:#{net.port.data}',
    }
  },

  deploy: {
    image: {"docker": "azukiapp/deploy-digitalocean"},
    mounts: {
      "/azk/deploy/src" :    path("."),
      "/azk/deploy/.ssh":    path("#{env.HOME}/.ssh"),
      "/azk/deploy/.config": persistent("deploy-config"),
    },
    scalable: {"default": 0, "limit": 0}
  }

});

```

**Comment by Bartekus** [comment]
I have not forgotten this, in-fact I've incorporated it into the build already however I'm still working on the details here and there so I need a little bit more time.

**Comment by Bartekus** [comment]
Ok your changes have been implemented and tested ok on the [Araphel](https://github.com/Araphel/meatier) branch

**Comment by mattkrick** [comment]
Awesome, I think the time might be near to merge the stacks and make this
*the* stack. Thanks for taking lead on this @bartekus!

**Comment by Bartekus** [comment]
My pleasure @mattkrick I'll be submitting a proper pull shortly and indeed, no better time than now and no better stack than meatier :D

**Comment by wmertens** [comment]
While this is cool functionality, wouldn't it be better to reuse an existing config file package, like https://github.com/lorenwest/node-config ? It has much more configurability and full tests for everything…

**Comment by mattkrick** [comment]
@wmertens i'll be honest, i've never seen node-config, but if the whole npm community can rally around a standardized deploy process i'm all for that instead of reinventing the wheel. Things like env fall short depending on where you deploy, but if this is truly universal in the sense that it doesn't matter if i deploy to bare metal or PaaS,  I'm all for it. @patrickleet any opinions?

**Comment by Bartekus** [comment]
I'm looking into this as soon as possible, just have few test to make! I'll keep you guys posted!

**Comment by gouroujo** [comment]
You can also check this nice stack that have included nice deployment tools : https://github.com/FredericHeem/starhackit

**Comment by Bartekus** [comment]
Let's keep this focused. 
The issue [How to deploy in production? #67] issue is resolved as of #95
I recommend closing this ticket and reopening one ticket per each outstanding issue.

**Comment by patrickleet** [comment]
@mattkrick @Bartekus  I have updates for this to make the rethinkdb SSL cert configurable to support multiple environments (QA deployment, vs production), but not really sure where to put it, as this was never pulled and instead pieces of it were just taken and put into @Bartekus branch. It'd be nice to get some PRs closed so we can build off master.

**Comment by Bartekus** [comment]
@patrickleet feel free to jump on r3stack branch and apply your modifications there, I'll be glad to add them. Matt is busy atm so we're getting a little backlogged but it will pass, just a lot on everybodys plate atm.

**Comment by mattkrick** [comment]
@patrickleet @Bartekus I think the best path forward is to accept this PR & then from there we can build an npm script that either statically rewrites the localhost to azk, or we can add another conditional to check for a deploy environment. You guys see any problem with this?

**Comment by Bartekus** [comment]
That is an excellent idea @mattkrick, so I'm full support of this approach!

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "babel-loader": "6.2.1",
     "babel-plugin-add-module-exports": "0.1.2",
     "babel-plugin-react-transform": "2.0.0",
+    "babel-plugin-static-fs": "^1.1.0",
```
Comment (mattkrick) [inline package.json]: can we remove this? fs is only used on the server as far as I can see

**Comment by mattkrick** [comment]
@patrickleet thanks for the PR & sorry it took so long for review! Aside from the extra babel plugin, it looks really solid. If you can patch that one line I'll merge tonight & then we can start the SSL fun!

**Comment by patrickleet** [comment]
I actually removed it already in the process of making it configurable.
Will submit a PR soon, in the midst of a 30something Hour return trip from
Cape Town.

**Comment by mattkrick** [comment]
no worries, i was in that same boat a couple days ago. I'm gonna hack on this for a little bit & I'll get a new master branch up soon so we can build off of that

**Pull request merged**